### PR TITLE
Remove pkg/context dependency from NEG controller

### PR DESCRIPTION
**Pull request by freehan**
cc @swetharepakula

### CODE REVIEW

File: pkg/neg/controller_test.go [hunk 1]
```diff
@@ -119,49 +114,81 @@ var (
 )
 
 func newTestController(kubeClient kubernetes.Interface) *Controller {
-	backendConfigClient := backendconfigclient.NewSimpleClientset()
-	namer := namer_util.NewNamer(ClusterID, "")
-	dynamicSchema := runtime.NewScheme()
-	//dynamicSchema.AddKnownTypeWithName(schema.GroupVersionKind{Group: "networking.istio.io", Version: "v1alpha3", Kind: "List"}, &unstructured.UnstructuredList{})
+	//backendConfigClient := backendconfigclient.NewSimpleClientset()
```
Comment (swetharepakula) [inline pkg/neg/controller_test.go:117]: commented code can be deleted.

File: pkg/neg/controller.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ package neg
 import (
 	context2 "context"
```
Comment (swetharepakula) [inline pkg/neg/controller.go:20]: This no longer needs to be imported as `context2`

File: pkg/neg/controller_test.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ import (
 	context2 "context"
```
Comment (swetharepakula) [inline pkg/neg/controller_test.go:20]: this no longer needs to be imported as `context2`

**Review by swetharepakula — COMMENTED** [review]
Overall looks good. 
A couple areas where commented code can be deleted.
A couple small suggestions to clean up code and to stop importing the built in `context` package as `context2`. But these can be ignored and addressed later.

File: pkg/neg/controller.go [hunk 4]
```diff
@@ -19,6 +19,7 @@ package neg
...
 	"fmt"
+	svcnegclient "k8s.io/ingress-gce/pkg/svcneg/client/clientset/versioned"
```
Comment (rramkumar1) [inline pkg/neg/controller.go:22]: move this to the second import block

File: pkg/neg/controller.go [hunk 5]
```diff
@@ -106,8 +106,22 @@ type Controller struct {
 
 // NewController returns a network endpoint group controller.
 func NewController(
+	kubeClient kubernetes.Interface,
+	svcNegClient svcnegclient.Interface,
+	destinationRuleClient dynamic.NamespaceableResourceInterface,
+	kubeSystemUID types.UID,
+	ingressInformer cache.SharedIndexInformer,
```
Comment (rramkumar1) [inline pkg/neg/controller.go:113]: why not encapsulate informers and clients into separate structs?

Comment (freehan) [reply pkg/neg/controller.go:113]: I can follow up with a PR to refactor this into a separate `NegControllerContext`.

**Comment by freehan** [comment]
Done. Ready for another round.

**Review by swetharepakula — APPROVED** [review]

**Pull request merged**